src/legacy/core_plugins/kbn_vislib_vis_types/public/editors/gauge.tsx

```diff
@@ -0,0 +1,267 @@
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { VisOptionsProps } from 'ui/vis/editors/default';
+import { RangesParamEditor } from 'ui/agg_types/controls/ranges';
+import { SelectOption } from '../controls/select';
+import { SwitchOption } from '../controls/switch';
+import { TextInputOption } from '../controls/text_input';
+import { GaugeVisParams } from '../gauge';
+
+function GaugeOptions(props: VisOptionsProps<GaugeVisParams>) {
+  const { stateParams, setValue, vis, uiState } = props;
+  const [isCustomColors, setIsCustomColors] = useState(false);
+
+  useEffect(() => {
+    uiState.on('colorChanged', () => {
+      setIsCustomColors(true);
+    });
+  }, []);
+
+  const setGaugeValue = useCallback(
+    <T extends keyof GaugeVisParams['gauge']>(paramName: T, value: GaugeVisParams['gauge'][T]) =>
+      setValue('gauge', {
+        ...stateParams.gauge,
+        [paramName]: value,
+      }),
+    [stateParams.gauge]
+  );
+
+  const setGaugeType = useCallback(
+    (paramName: 'gaugeType', value: GaugeVisParams['gauge']['gaugeType']) => {
+      const minAngle = value === 'Arc' ? undefined : 0;
+      const maxAngle = value === 'Arc' ? undefined : 2 * Math.PI;
+
+      setValue('gauge', {
+        ...stateParams.gauge,
+        [paramName]: value,
+        minAngle,
+        maxAngle,
+      });
+    },
+    [stateParams.gauge]
+  );
+
+  const resetColorsButton = (
+    <EuiButtonEmpty
+      size="xs"
+      onClick={() => {
+        uiState.set('vis.colors', null);
+        setIsCustomColors(false);
+      }}
+    >
+      <FormattedMessage
+        id="kbnVislibVisTypes.controls.gaugeOptions.resetColorsButtonLabel"
+        defaultMessage="Reset colors"
+      />
+    </EuiButtonEmpty>
+  );
+
+  return (
+    <>
+      <EuiPanel paddingSize="s">
+        <EuiTitle size="xs">
+          <h2>
+            <FormattedMessage
+              id="kbnVislibVisTypes.controls.gaugeOptions.styleTitle"
+              defaultMessage="Style"
+            />
+          </h2>
+        </EuiTitle>
+        <EuiSpacer size="s" />
+
+        <SelectOption
+          label={i18n.translate('kbnVislibVisTypes.controls.gaugeOptions.gaugeTypeLabel', {
+            defaultMessage: 'Gauge type',
+          })}
+          options={vis.type.editorConfig.collections.gaugeTypes}
+          paramName="gaugeType"
+          value={stateParams.gauge.gaugeType}
+          setValue={setGaugeType}
+        />
+
+        <SwitchOption
+          label={i18n.translate('kbnVislibVisTypes.controls.gaugeOptions.percentageModeLabel', {
+            defaultMessage: 'Percentage mode',
+          })}
+          paramName="percentageMode"
+          value={stateParams.gauge.percentageMode}
+          setValue={setGaugeValue}
+        />
+        <EuiSpacer size="s" />
+
+        <SelectOption
+          label={i18n.translate('kbnVislibVisTypes.controls.gaugeOptions.alignmentLabel', {
+            defaultMessage: 'Alignment',
```

Comment:
Disable this one if there is only a single Gauge chart

Reply:
Hi @cchaos 
Number of Gauges on the chart depends on buckets size which return from elasticsearch.
On the UI we are only available to restrict the maximum size of the buckets -> set `size` in terms ->

![image](https://user-images.githubusercontent.com/31325372/63158779-b0d9de00-c022-11e9-8024-305b1fa277ed.png)

The `size` doesn't guarantee number of gauges on the chart, just limit its number!
So on the UI side we can't know beforehand how many gauges will be shown. It will depend on the returning data (due to selected time range).
Take a look at the gif below:



![gauge](https://user-images.githubusercontent.com/31325372/63159274-da473980-c023-11e9-83a9-54728d39f68f.gif)

Reply:
Yep, I understand. I guess I'm trying to say not necessarily what's being rendered. But that there **is a split group** applied.

Reply:
@sulemanof Did you see my response above ^^?

Reply:
ok, done

Reply:
Great, thank you!